lib/puppet/pops/issues.rb

```diff
@@ -130,7 +135,7 @@ def self.hard_issue(issue_code, *args, &block)
   # @todo configuration
   #
   NAME_WITH_HYPHEN = issue :NAME_WITH_HYPHEN, :name do
-    "#{label.a_an_uc(semantic)} may not have a name containing a hyphen. The name '#{name}' is not legal"
+    "#{provider.a_an_uc(semantic)} may not have a name containing a hyphen. The name '#{name}' is not legal"
```

Comment:
The "provider.a_an_uc" does not read nicely, hence my suggestion to get rid of it completely. I think that is acceptable to do within the scope of this ticket. It is a very small additional change.